Kick output vectors

reinhold-willcox opened the pull request:
Added post-SN orbital AM vector (in pre-SN frame of reference) to the possible BINARY_PROPERTIES that can be included in the output. The parameters are 

`SUPERNOVA_ORBIT_INCLINATION_VECTOR_X`, 
`SUPERNOVA_ORBIT_INCLINATION_VECTOR_Y`, and 
`SUPERNOVA_ORBIT_INCLINATION_VECTOR_Z`. 

They are only useful in some rare use cases, so they are not included in the default `BSE_SUPERNOVAE` logfile, but can be added with a custom `logfile-definitions` file.

Separately, added documentation for the following properties, addressing issue #821: 
```
RLOF_POST_STEP_STAR_TO_ROCHE_LOBE_RADIUS_RATIO_1
RLOF_POST_STEP_STAR_TO_ROCHE_LOBE_RADIUS_RATIO_2
RLOF_PRE_STEP_STAR_TO_ROCHE_LOBE_RADIUS_RATIO_1
RLOF_PRE_STEP_STAR_TO_ROCHE_LOBE_RADIUS_RATIO_2
```

reinhold-willcox committed:
added member variable for post-SN orbital AM vector, with capability to include in SN output (excluded by default)

reinhold-willcox committed:
added radiusStarX/RL options to documentation, addressing issue #821

reinhold-willcox committed:
added X,Y,Z-components of post-SN orbital AM vector to documentation

reinhold-willcox committed:
Merge branch 'dev' into kick_output_vectors

reinhold-willcox committed:
changelog

ilyamandel reviewed (changes requested):
Thanks for the changes, @reinhold-willcox .  A few queries/thoughts:

- In the documentation, it would be great to have more information about the angular momentum vector.  E.g., how is the frame defined (is \hat{z} along the pre-SN orbital along momentum vector?  what about \hat{x}?)?  What are the units (I see in the code that you've normalised the vector, but I wouldn't know that without looking at the code... and if it is normalized, why not just output two angles, theta and phi?)?

- Should the label be something like Orbit_Inclination_Vector_Z>SN rather than SN_Orbit_Inclination_Vector_Z?

- Vector3d m_OrbitalAngularMomentumVector; // Systemic velocity vector, relative to ZAMS Center of Mass
I think this comment in BaseBinaryStar.h is incorrect.

reinhold-willcox committed:
fixed description of m_OrbitalAngularMomentumVector

reinhold-willcox committed:
updated label Orbit_Inclination_Vector_N>SN to Orbital_AM_Vector>SN_N where _N is any of _X, _Y, _Z

reinhold-willcox committed:
cleaned up description of the new post-SN orbital AM vector components

reinhold-willcox commented:
@ilyamandel I've now addressed your comments. Regarding the unit vector components vs two new angles, I chose the former simply because it's more directly useful for my project with Ryo, and the code already has the vector component information readily stored (so I don't have to define two new angles and then write up documentation about them), though I agree that the latter is slightly more storage efficient. If we decide these parameters are likely to be useful in a broader context, it may be worth discussing which is the better approach.

ilyamandel committed:
Update standard-logfiles-record-specification-binary.rst

Making it clear that we are dealing with dimensionless unit vector components.

ilyamandel reviewed (approved):
All good.
